src/Modules/UngradedModules.jl

```diff
@@ -226,7 +332,31 @@ function show(io::IO, e::FreeModElem)
     print(io, 0)
     return
   end
-  print(io, join(["($val)*$(e.parent.S[pos])" for (pos, val) in e.coords], " + "))
+  out_string = join(["($val)*$(e.parent.S[pos])" for (pos, val) in e.coords], " + ")
+  out_string = replace(out_string, "(1)*"=>"")
```

Comment:
This is fragile, though: what if the underlying ring prints something that contains the substring `(1)*` but which must not be removed like that?

I think this either require slightly less lazy loop to generate the right string from the start (it's not that complicated, is it); or perhaps the pretty printing code by @tthsqe12 I've heard about can handle it?

Reply:
Sure, this printing code is not done right. Before expressify, this would have been done "properly" with `displayed_with_minus_in_front` and `is_negative`. Now the easiest and most robust way is
```
function expressify(e::FreeModElem; context = nothing)
   sum = Expr(:call, :+)
   for (pos, val) in e.coords
      # assuming e.parent.S is an array of strings/symbols
      push!(sum, Expr(:call, :*, expressify(val, context = context), e.parent.S[pos]))
   end
   return sum
end

@enable_all_show_via_expressify FreeModElem
```

Reply:
Thanks Dan! That seems quite nice!